File: ruby/lib/task_master/output_handlers/task_summary_report/terminal_handler.rb
module TaskMaster
  module OutputHandlers
    module TaskSummaryReport
      class TerminalHandler
        def generate_output(data)
          AsciiTable.new(headers, DataProcessor.new(data).process).draw
        end

        private

        def headers
          ["Metric", "Count"]
        end

        class DataProcessor
          def initialize(data)
            @data = data
          end

          def process
            [
              ["Total Tasks", @data.count.to_s],
              ["Completed Tasks", @data.count(&:complete?).to_s],
              ["Tasks in Progress", @data.count(&:in_progress?).to_s],
              ["Tasks Not Started", @data.count(&:not_started?).to_s]
            ]
          end
        end
      end
    end
  end
end

File: javascript/lib/outputHandlers/taskSummaryReport/TerminalHandler.js
const path = require('path');
const AsciiTable = require(path.join(__dirname, '../../AsciiTable'));

class TerminalHandler {
	constructor() {
		this.AsciiTable = AsciiTable
		this.DataProcessor = class {
			constructor(data) {
				this.data = data
			}

			process() {
				return [
					["Total Tasks", this.data.length.toString()],
					["Completed Tasks", this.data.filter((task) => task.isComplete()).length.toString()],
					["Tasks in Progress", this.data.filter((task) => task.isInProgress()).length.toString()],
					["Tasks Not Started", this.data.filter((task) => task.isNotStarted()).length.toString()],
				]
			}

			#formatDateString(date) {
				return String(date.getFullYear() + '-' + (date.getMonth() + 1) + '-' + String(date.getDate()).padStart(2, '0'));
			}
		}
	}

	generateOutput(data) {
		return new this.AsciiTable(this.#headers(), new this.DataProcessor(data).process()).draw()
	}

	#headers() {
		return ["Metric", "Count"]
	}
}

module.exports = TerminalHandler;

